orchestrator/careplanservice/handle_get.go

```diff
@@ -0,0 +1,191 @@
+package careplanservice
+
+import (
+	"context"
+	"encoding/json"
+	fhirclient "github.com/SanteonNL/go-fhir-client"
+	"github.com/SanteonNL/orca/orchestrator/lib/auth"
+	"github.com/SanteonNL/orca/orchestrator/lib/coolfhir"
+	"github.com/rs/zerolog/log"
+	"github.com/zorgbijjou/golang-fhir-models/fhir-models/fhir"
+	"net/http"
+	"strings"
+)
+
+func (s *Service) handleGet(httpResponse http.ResponseWriter, request *http.Request) error {
+	queryParams := request.URL.Query()
+	urlParts := strings.Split(strings.TrimPrefix(request.URL.Path, basePath+"/"), "/")
+	switch urlParts[0] {
+	case "CarePlan":
```

Comment:
This is actually a search, might be more clear if you have separate routing for the two. E.g.:

```go
// Handle reading a specific resource instance
mux.HandleFunc("GET "+basePath+"/{type}/{id}", ...)
// Handle search
mux.HandleFunc("GET "+basePath+"/{type}", ...)
```

or specific for the resource type;

```go
```go
// Handle reading a specific resource instance
mux.HandleFunc("GET "+basePath+"/CarePlan/{id}", ...)
// Handle search
mux.HandleFunc("GET "+basePath+"/CarePlan", ...)
```

The latter is more like how the create funcs are working
```

Reply:
I have implemented this for the next revision